src/pkcs15init/pkcs15-lib.c

```diff
@@ -1294,7 +1295,7 @@ sc_pkcs15init_init_skdf(struct sc_pkcs15_card *p15card, struct sc_profile *profi
 	struct sc_pkcs15_object *object = NULL;
 	const char	*label;
 	unsigned int	usage;
-	unsigned int	keybits = keyargs->value_len;
+	unsigned int	keybits = keyargs->value_len * 8;
```

Comment:
has this been incorrect before?

Reply:
The issue is that CKA_VALUE_LEN that comes in a PKCS#11 template means number of bytes, but in a PKCS#15 SecretKeyObject, the key length should be set in bits.

After looking at pkcs15-init tool and framework-pkcs15, I changed it so that the multiplication to bits is done in framework-pkcs15, before passing keyargs to pkcs15-lib, and removed the multiplication from pkcs15-lib.

pkcs15-init tool passes length in bits, and the functionality to store secret keys with the tool is already in the release and may be used by somebody. So I think it's best to be consistent with that.

I also changed line 2356 of framework-pkcs15.c to multiply key length by 8, where software-only secret key objects are created. At least I couldn't find any side effects this could cause.  This caused the confusion in the first place: framework-pkcs15 was setting sc_pkcs15_skey_info.value_len in bytes and pkcs15-lib was setting the same member of the same struct in bits.